### PR TITLE
storage/zynq7000/qspi: configure qspi pins based on board_config data

### PR DESCRIPTION
**Pull request by HBuczynski**
## Description

PP-97

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/zynq7000-flash/qspi.c [hunk 1]
```diff
@@ -382,16 +393,21 @@ static int qspi_initPins(void)
...
-		pctl_mio_pin_05, /* I/O */
-		pctl_mio_pin_06, /* CLK */
-		pctl_mio_pin_08  /* Feedback CLK */
+		QSPI_CS,
+		QSPI_IO0,
+		QSPI_IO1,
+		QSPI_IO2,
+		QSPI_IO3,
+		QSPI_CLK,
+		QSPI_FCLK
```
Comment (agkaminski) [inline storage/zynq7000-flash/qspi.c:402]: Shouldn't we translate these defines (integers) to pctl_mio_pins somehow?

File: storage/zynq7000-flash/qspi.c [hunk 2]
```diff
@@ -294,9 +295,19 @@ static void qspi_IOMode(void)
 
 	/* Configure controller */
 
-	/* Set baud rate to 100 MHz: 200 MHz / 2, master mode, not Legacy mode */
+	/* Set master mode, not Legacy mode */
+	*(qspi_common.base + cr) = 0x1 | (1 << 31);
+
 	*(qspi_common.base + cr) &= ~(0x7 << 3);
-	*(qspi_common.base + cr) |= 0x1 | (1 << 31);
+	if (QSPI_FCLK != -1) {
```
Comment (agkaminski) [inline storage/zynq7000-flash/qspi.c:302]: Just my OCD, any negative value is invalid
```suggestion
	if (QSPI_FCLK >= 0) {
```

File: storage/zynq7000-flash/qspi.c (hunk 1 above)
Comment (HBuczynski) [reply storage/zynq7000-flash/qspi.c:402]: I think that we decided that there should no be any translation. Look into other drivers which use `board_config.h`

Comment (agkaminski) [reply storage/zynq7000-flash/qspi.c:399]: OK

**Review by agkaminski — APPROVED** [review]

**Pull request merged**